server/routerlicious/packages/services-core/src/runWithRetry.ts

```diff
@@ -88,20 +88,20 @@ export async function runWithRetry<T>(
  * anything other than a `T` value - the only other possibility is a promise rejection.
  * @param  {()=>Promise<T>} request - function to run and retry in case of error
  * @param  {string} callName - name of the api function we are calling
+ * @param  {(error)=>boolean} shouldRetry - function that takes error and decides whether to retry on it
  * @param  {number} maxRetries - maximum retries after which error is thrown. Retry infinitely if set to -1
  * @param  {number} retryAfterMs - interval factor to wait before retrying. Param to calculateIntervalMs
- * @param  {(error)=>boolean} shouldRetry? - function that takes error and decides whether to retry on it
  * @param  {(error, numRetries, retryAfterInterval)=>number} calculateIntervalMs
  * function which calculates interval to wait before retrying based on error, retryAfterMs and retries so far
  * @param  {(error)=>void} onErrorFn? - function allowing caller to define custom logic to run on error e.g. custom logs
  */
  export async function requestWithRetry<T>(
     request: () => Promise<T>,
     callName: string,
-    maxRetries: number,
-    retryAfterMs: number,
-    shouldRetry?: (error) => boolean,
-    calculateIntervalMs = (error, numRetries, retryAfterInterval) => retryAfterInterval * 2 ** numRetries,
+    shouldRetry: (error) => boolean = shouldRetryNetworkError,
+    maxRetries: number = -1,
+    retryAfterMs: number = 1000,
+    calculateIntervalMs = calculateRetryIntervalForNetworkError,
```

Comment:
Does this need a type? Like `calculateIntervalMs: (error: any, numRetries: number, retryAfterInterval: number) => number`?

Reply:
Good point - I removed the type by accident while adding the default value. Let me fix it.